runtime/src/dispatcher.rs

```diff
@@ -419,7 +422,7 @@ impl Dispatcher {
                 max_messages,
                 method,
                 args,
-            } => {
+            } if state.txn_dispatcher.is_supported() => {
```

Comment:
Not that pretty :thinking:

Reply:
Indeed, this should all be refactored to have some sort of App trait that covers both variants. But this seemed like something that would be a bigger effort in itself so I didn't want to do it in this PR.